$NetBSD: patch-CMakeLists.txt,v 1.5 2026/07/30 12:36:00 gdt Exp $

CMake will detect the MacOS 6-argument sendfile() support and configure
HAVE_SENDFILE, but the code itself doesn't support this interface.  So don't
probe for HAVE_SENDFILE if we HAVE_SENDFILE6_SUPPORT.

Just because librt doesn't exist does not mean that clock_gettime() doesn't
exist, so don't insist on librt being present.  Fixes the build on MacOS.

Let pkgsrc be responsible for zm.conf into examples, rather than overwriting
the existing configuration file.

--- CMakeLists.txt.orig	2026-05-28 12:42:49.000000000 +0000
+++ CMakeLists.txt
@@ -329,7 +329,9 @@ check_include_file("ucontext.h" HAVE_UCO
 check_include_file("sys/sendfile.h" HAVE_SYS_SENDFILE_H)
 check_include_file("sys/syscall.h" HAVE_SYS_SYSCALL_H)
 check_function_exists("syscall" HAVE_SYSCALL)
+if(NOT HAVE_SENDFILE6_SUPPORT)
 check_function_exists("sendfile" HAVE_SENDFILE)
+endif(NOT HAVE_SENDFILE6_SUPPORT)
 check_function_exists("posix_memalign" HAVE_POSIX_MEMALIGN)
 check_type_size("siginfo_t" HAVE_SIGINFO_T)
 check_type_size("ucontext_t" HAVE_UCONTEXT_T)
@@ -741,7 +743,7 @@
 # Check all required modules at once
 # TODO: Add checking for the optional modules
 find_package(
-  PerlModules COMPONENTS Sys::Syslog DBI DBD::mysql
+  PerlModules COMPONENTS Sys::Syslog DBI DBD::MariaDB
   Getopt::Long Time::HiRes Date::Manip LWP::UserAgent
   ExtUtils::MakeMaker ${ZM_MMAP_PERLPACKAGE})
 if(NOT PERLMODULES_FOUND)
@@ -839,8 +841,8 @@ else()
 endif()
 
 # Install zm.conf
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}/zm.conf" DESTINATION "${ZM_CONFIG_DIR}")
-install(DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/conf.d/" DESTINATION "${ZM_CONFIG_SUBDIR}" PATTERN "*.in" EXCLUDE)
+#install(FILES "${CMAKE_CURRENT_BINARY_DIR}/zm.conf" DESTINATION "${ZM_CONFIG_DIR}")
+#install(DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/conf.d/" DESTINATION "${ZM_CONFIG_SUBDIR}" PATTERN "*.in" EXCLUDE)
 
 # Uninstall target
 configure_file(
